s3_upload/utils/upload.py

```diff
@@ -134,15 +134,62 @@ def upload_single_file(
 
     log.debug("%s uploaded as %s", local_file, remote_object.get("ETag"))
 
-    return local_file, remote_object.get("ETag").strip('"')
+    return local_file, remote_object.get("ETag", "").strip('"')
+
+
+def _submit_to_pool(pool, func, item_input, items, **kwargs):
```

Comment:
Does this not need testing as well or is it incorporated in `multi_thread_upload()` and `multi_core_upload()` testing?